src/components/component/secrets/secret-list-item.tsx

```diff
@@ -28,7 +28,7 @@ export const SecretListItem = ({
 }): JSX.Element => {
   const [isExpanded, setIsExpanded] = useState(false);
 
-  const hasStatusMessages = secret.statusMessages?.length > 0;
+  const hasStatusMessages = secret?.statusMessages?.length > 0;
```

Comment:
secret is not an optional input, parent/caller have responsibility to null check
```suggestion
  const hasStatusMessages = secret.statusMessages?.length > 0;
```